xen/arch/arm/coproc/coproc.h

```diff
@@ -167,6 +165,30 @@ static inline void vcoproc_get_rw_context(struct domain *d, struct mmio *mmio,
     BUG_ON(ctx->vcoproc == NULL);
 }
 
+/* time slice for this vcoproc has finished, save context and pause:
+ * return value:
+ *    0 - on success
+ *   >0 - if more time is needed: time in ms that vcoproc still needs
+ *        before next attempt to switch context should be made
+ *   <0 - unrecoverable failure: no future context switches are possible,
+ *        coproc must be considered as non-functional
+ */
+static inline s_time_t vcoproc_context_switch_from(struct vcoproc_instance *vcoproc)
+{
+    return vcoproc->coproc->ops->ctx_switch_from(vcoproc);
+}
+
+/* new vcoproc has time slice, restore context and unpause:
+ * return value:
+ *  >=0 - on success
```

Comment:
We have a definition of function, so it is absolutely author's right to limit return value to `(-&infin; ; 0]` and so describe only those values.
```c
+ * return value:
+ *  0 - no error
+ *  <0 - error code
```
BTW, do we use standard error codes here? Just wondering.